libs/isograph-react/src/core/reader.ts

```diff
@@ -144,7 +144,7 @@ type StableId = string;
 export type LoadableField<
   TReadFromStore extends { data: object; parameters: object },
   TResult,
-  TProvidedArgs extends object = object,
+  TArgs = ExtractParameters<TReadFromStore>,
```

Comment:
So I've this to have more control, now `useSkipLimitPagination` can say it takes
```tsx
type UseSkipLimitPaginationArgs = {
  skip: number;
  limit: number;
};
```
and everything will be type checked nicely

Reply:
nice, what was not type checking before?

Reply:
This wasn't working without `@ts-expect-error`
https://github.com/isographlabs/isograph/blob/6bd3135f11b67c408bac8ec5118cf5e0a598049e/libs/isograph-react/src/loadable-hooks/useSkipLimitPagination.ts#L162-L167
Typescript said `skip` and `limit` were not existing, although types where
```tsx
(parameter) loadableField: (args: void |
Omit<TReadFromStore["parameters"], keyof
Omit<TReadFromStore["parameters"], "skip" | "limit">>) => [...]
```
and `TReadFromStore["parameters"]` extended `UseSkipLimitPaginationArgs`, so it should have, `skip` and `limit` keys, somehow `Omit` made it impossible to infer the type.
But this wasn't a big deal.

## Bigger issue was Covariance
Now the parameters are `covariant`.
If user tries to pass `LoadableField` that somehow takes
```tsx
{
  skip: number | string | SomeOther,
  limit: number | null
}
```
(creates a wrapper, have custom scalar or widens types?) 

then it's not an issue, because `number` is assignable to `number | string | SomeOther`.

Previously `parameters` would be used in a return type of `loadable`; as a result type of `parameters` was `contravariant`.
We would have to predict all the possible values `skip` and `limit` can take
```tsx
interface UseSkipLimitPaginationArgs {
  skip: number | null | string | AnyOther,
  limit: number | null
}
```
Inside of `useSkipLimitPagination` we don't care directly about the type of parameters returned by `loadableField`.
By splitting `parameters` into arguments and return type in `LoadableField` we can have `contravariant` return type and `covariant` arguments.

With the default `TArgs = ExtractParameters<TReadFromStore>`, none of the call sites had to be changed.
But when we want to provide arguments, the logic have to be done one level higher, that's why change in rust has been required.
```rs
format!(
    ",\n{indent}Omit<ExtractParameters<{}__param>, keyof {}>",
    client_field.type_and_field.underscore_separated(),
    get_loadable_field_type_from_arguments(
        schema,
        provided_arguments
    )
)
```

Reply:
Okay, awesome, thanks for digging into that and figuring this out!!

> Previously parameters would be used in a return type of loadable;

where? Is that because parameters was returned as part of `FragmentReference`?

Reply:
Yes, the fragment includes parameters